Tests/SwiftParserTest/DeclarationTests.swift

```diff
@@ -760,7 +760,7 @@ final class DeclarationTests: XCTestCase {
       }
       """,
       diagnostics: [
-        DiagnosticSpec(message: "unexpected code 'bogus rethrows set' in variable")
+        DiagnosticSpec(message: "unexpected code 'bogus' before effect specifiers")
       ]
     )
   }
```

Comment:
```
func foo() bogus async {}
func foo() throws bogus {}
func foo() bogus -> Int {}
```

If I'm reading the code correctly I'm fairly sure they aren't handled well. It'd be good to add + fix these, but also happy to have that in a separate PR if you want though, since it's no worse than what's there at the moment.

Reply:
The first one is handled well. For the other two, I’ll look into them in a follow-up PR.